nautobot/dcim/forms.py

```diff
@@ -264,6 +264,25 @@ def clean(self):
                 )
 
 
+class ModularComponentForm(ComponentForm):
+    name_pattern = ExpandableNameField(
+        label="Name",
+        help_text="""
+            Alphanumeric ranges are supported for bulk creation. Mixed cases and types within a single range
+            are not supported. Examples:
+            <ul>
+                <li><code>[ge,xe]-0/0/[0-9]</code></li>
+                <li><code>e[0-3][a-d,f]</code></li>
+            </ul>
+
+            The variables <code>{module}</code>, <code>{module.parent}</code>, <code>{module.parent.parent}</code>, etc.
+            may be used in the name field and will be replaced by the position that the module occupies in the parent
+            device or module. These variables can be used multiple times in the component name and there is no limit to
+            the depth of parent levels. Any variables that cannot be replaced by a suitable parent position will remain unchanged.
```

Comment:
Aren't we limiting the modules themselves to a depth of 3/4?

Reply:
We're just limiting how many levels we'll recurse to 3/4 for displaying components in the UI. As far as the model is concerned, as long as you don't create an infinite loop you can go further than that.